syntax = "proto3";

package proto;

service GossipService {
  rpc Increment (GossipMessage) returns (IncrementResponse) {}
  rpc GetGossipCount (GossipMessage) returns (GetCountResponse) {}
  // TODO: Add a method to communicate with other nodes
}

message GossipMessage {}

message IncrementResponse {
  bool success = 1;
}

message GetCountResponse {
  int64 count = 1;
}
